src/modals/cpu-pinning/CpuPinningDataProvider.js

```diff
@@ -83,47 +111,93 @@ const getPlacementPolicyHostsIds = (vm) => {
   return hostsIds
 }
 
-const getCpus = (host) => {
-  const topology = host.cpu?.topology
-  if (topology) {
-    return topology.sockets * topology.cores * topology.threads
-  }
-  return 0
-}
-
-const mapVmToModel = (vm) => {
-  return {
-    id: vm.id,
-    name: vm.name,
-    cpuTopology: {
-      sockets: parseInt(getSockets(vm)),
-      cores: parseInt(getCores(vm)),
-      threads: parseInt(getThreads(vm)),
-    },
-    cpuPinningPolicy: vm.cpu_pinning_policy,
-    cpuPinnings: getPinnings(vm),
+const mapVmToPinnedEntity = (vm) => {
+  const pinned = new PinnedEntity(vm.id, vm.name)
+
+  const sockets = +getSockets(vm)
+  const cores = +getCores(vm)
+  const threads = +getThreads(vm)
+  const cpuToPinnedCpuMap = getCpuToPinnedCpuMapForVm(vm)
+
+  if (sockets && cores && threads) {
+    for (let s = 0; s < +sockets; s++) {
+      for (let c = 0; c < +cores; c++) {
+        for (let t = 0; t < +threads; t++) {
+          const cpuId = s * cores * threads + c * threads + t
```

Comment:
please describe (a comment in the code) why/how we are calculating `cpuId`

Reply:
you're right, it seems complicated - I've changed to simply increment a value for every iteration.